src/components/Timer.tsx

```diff
@@ -13,25 +13,35 @@ export default function Timer(props: { index: number }) {
     const index = props.index;
     const { q_index } = useAppSelector(state => state.time);
     let { targetTime, secondsLeft } = useAppSelector(state => state.time);
-    const { rejoinAsHost } = useAppSelector(state => state.player);
-    let rejoined = rejoinAsHost;
+    const { rejoinAsHost, rejoinAsPlayer, isHost } = useAppSelector(state => state.player);
+    let rejoined = rejoinAsHost || rejoinAsPlayer;
 
     // if new question, start timer by creating new target time
     if (index !== q_index && index !== undefined) {
         const timeNow = new Date().getTime();
         targetTime = timeNow + TIME_LIMIT_IN_MS;
-        // Check if the host has just rejoined the game
+        // Check if the host/player has just rejoined the game
         if (rejoined) {
             dispatch(setRejoinAsHost(false));
+            dispatch(setRejoinAsPlayer(false));
             rejoined = false;
-            targetTime = timeNow + secondsLeft*1000;
+            const indexCookie = Cookies.get('index');
+            // If we are rejoining as a host OR if we are rejoining as a player at the question we
+            // were at previously, use the stored secondsLeft for the targetTime (coming from the cookies)
+            if (isHost || (!isHost && indexCookie !== undefined && (Number(indexCookie)) === index)) {
+                // targetTime = timeNow + TIME_LIMIT_IN_MS;
+                targetTime = timeNow + secondsLeft * 1000;
+            }
+            dispatch(setRejoinAsHost(false));
+            dispatch(setRejoinAsPlayer(false));
```

Comment:
Is there a reason why you have this here when you already set these flags to `False` in lines 25-26?